arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/concurrent/PubSub.kt

```diff
@@ -79,15 +78,15 @@ internal interface Subscribe<A, Selector> {
 
 internal interface PubSub<I, O, Selector> : Publish<I>, Subscribe<O, Selector> {
 
-  data class Publisher<A>(val token: Token, val i: A, val signal: Promise<Unit>) {
-    suspend fun complete(): Unit {
-      ForkConnected { signal.complete(Unit) }
```

Comment:
It's not necessary to launch completion in a `Fiber` since the `join` operations already `intercept` before returning, meaning they always get properly re-scheduled before returning from `join`.